### ISSUE TITLE
结果复现

### ISSUE THREAD

**Issue by lpc-97** [hitachinsk/SAMed#55]
parser = argparse.ArgumentParser()
parser.add_argument('--root_path', type=str,
                    default='/data/LarryXu/Synapse/preprocessed_data/train_npz', help='root dir for data')
parser.add_argument('--output', type=str, default='/output/sam/results')
parser.add_argument('--dataset', type=str,
                    default='Synapse', help='experiment_name')
parser.add_argument('--list_dir', type=str,
                    default='./lists/lists_Synapse', help='list dir')
parser.add_argument('--num_classes', type=int,
                    default=8, help='output channel of network')
parser.add_argument('--max_iterations', type=int,
                    default=30000, help='maximum epoch number to train')
parser.add_argument('--max_epochs', type=int,
                    default=200, help='maximum epoch number to train')
parser.add_argument('--stop_epoch', type=int,
                    default=160, help='maximum epoch number to train')
parser.add_argument('--batch_size', type=int,
                    default=8, help='batch_size per gpu')
parser.add_argument('--n_gpu', type=int, default=1, help='total gpu')
parser.add_argument('--deterministic', type=int, default=1,
                    help='whether use deterministic training')
parser.add_argument('--base_lr', type=float, default=0.005,
                    help='segmentation network learning rate')
parser.add_argument('--img_size', type=int,
                    default=512, help='input patch size of network input')
parser.add_argument('--seed', type=int,
                    default=1234, help='random seed')
parser.add_argument('--vit_name', type=str,
                    default='vit_b', help='select one vit model')
parser.add_argument('--ckpt', type=str, default='checkpoints/sam_vit_b_01ec64.pth',# sam_vit_b_01ec64 medsam_vit_b
                    help='Pretrained checkpoint')
parser.add_argument('--lora_ckpt', type=str, default=None, help='Finetuned lora checkpoint')
parser.add_argument('--rank', type=int, default=4, help='Rank for LoRA adaptation')
parser.add_argument('--warmup', action='store_true', help='If activated, warp up the learning from a lower lr to the base_lr')
parser.add_argument('--warmup_period', type=int, default=250,
                    help='Warp up iterations, only valid whrn warmup is activated')
parser.add_argument('--AdamW', action='store_true', help='If activated, use AdamW to finetune SAM model')
parser.add_argument('--module', type=str, default='sam_lora_image_encoder')
parser.add_argument('--dice_param', type=float, default=0.8)
args = parser.parse_args()

——

Namespace(config=None, volume_path='testset/test_vol_h5/', dataset='Synapse', num_classes=8, list_dir='./lists/lists_Synapse/', output_dir='/mnt/d/Linux/SAMed-main/output', img_size=512, input_size=224, seed=1234, is_savenii=True, deterministic=1, ckpt='checkpoints/sam_vit_b_01ec64.pth', lora_ckpt='checkpoints/epoch_159.pth', vit_name='vit_b', rank=4, module='sam_lora_image_encoder')
12 test iterations per epoch
0it [00:00, ?it/s]idx 0 case case0008 mean_dice 0.441581 mean_hd95 34.595021
1it [02:15, 135.34s/it]idx 1 case case0022 mean_dice 0.684595 mean_hd95 3.275303
2it [03:32, 101.11s/it]idx 2 case case0038 mean_dice 0.669412 mean_hd95 27.522629
3it [05:01, 95.56s/it] idx 3 case case0036 mean_dice 0.681733 mean_hd95 19.580403
4it [07:59, 128.04s/it]idx 4 case case0032 mean_dice 0.621131 mean_hd95 6.248785
5it [10:06, 127.68s/it]idx 5 case case0002 mean_dice 0.690973 mean_hd95 2.605073
6it [12:07, 125.58s/it]idx 6 case case0029 mean_dice 0.671065 mean_hd95 18.496830
7it [13:33, 112.38s/it]idx 7 case case0003 mean_dice 0.428255 mean_hd95 71.256974
8it [16:39, 135.90s/it]idx 8 case case0001 mean_dice 0.653609 mean_hd95 26.554855
9it [18:54, 135.68s/it]idx 9 case case0004 mean_dice 0.666332 mean_hd95 14.264169
10it [20:58, 132.16s/it]idx 10 case case0025 mean_dice 0.639110 mean_hd95 7.775622
11it [22:11, 113.85s/it]idx 11 case case0035 mean_dice 0.662067 mean_hd95 7.006532
12it [23:28, 117.42s/it]
Mean class 1 name spleen mean_dice 0.863648 mean_hd95 9.370867
Mean class 2 name right kidney mean_dice 0.805086 mean_hd95 38.663117
Mean class 3 name left kidney mean_dice 0.804298 mean_hd95 42.883881
Mean class 4 name gallbladder mean_dice 0.000000 mean_hd95 0.000000
Mean class 5 name liver mean_dice 0.922219 mean_hd95 36.088610
Mean class 6 name stomach mean_dice 0.763024 mean_hd95 23.126339
Mean class 7 name aorta mean_dice 0.848302 mean_hd95 9.321985
Mean class 8 name pancreas mean_dice 0.000000 mean_hd95 0.000000
Testing performance in best val model: mean_dice : 0.625822 mean_hd95 : 19.931850
Testing Finished!
————

结果复现 无法达到你提供的权重的结果 我自己重新训练的

**Comment by Rilomilo** [comment]
#44 你可以参考一下这一条issue
